daemon/config.go

```diff
@@ -75,6 +75,8 @@ func (config *Config) InstallFlags() {
 	config.Ulimits = make(map[string]*ulimit.Ulimit)
 	opts.UlimitMapVar(config.Ulimits, []string{"-default-ulimit"}, "Set default ulimits for containers")
 	flag.StringVar(&config.LogConfig.Type, []string{"-log-driver"}, "json-file", "Default driver for container logs")
+	config.LogConfig.Config = make(map[string]string)
```

Comment:
this could be put in the newMapOpt constructor. Just check if map is nil, and create it if it's nil in the constructor.